nav2_amcl/src/amcl_node.cpp

```diff
@@ -75,6 +76,11 @@ AmclNode::AmclNode()
   declare_parameter("laser_max_range", rclcpp::ParameterValue(100.0));
   declare_parameter("laser_min_range", rclcpp::ParameterValue(-1.0));
   declare_parameter("laser_model_type", rclcpp::ParameterValue(std::string("likelihood_field")));
+  declare_parameter("set_initial_pose", rclcpp::ParameterValue(false));
+  declare_parameter("initial_pose.x", rclcpp::ParameterValue(0.0));
```

Comment:
For now, It's okay to have 4 different parameters to define pose since I don't think declace_parameter currently supports Pose msg. We should consider thinking about how to use a custom msg as a parameter.

Reply:
Yes, that would be a nice feature.